File: leetcode/array/minimum-size-subarray-sum.go
// https://leetcode.com/problems/minimum-size-subarray-sum
func minSubArrayLen(target int, nums []int) int {
	s := len(nums)
	j := s + 1
	arr := make([]int, s+1)

	for i := 1; i <= s; i++ {
		arr[i] = arr[i-1] + nums[i-1]
	}

	for l := 0; l < s; l++ {
		for r := l; r < s; r++ {
			if arr[r+1]-arr[l] >= target {
				if r-l+1 < j {
					j = r - l + 1
				}
			}
		}
	}

	if j > s {
		return 0
	}
	return j
}

File: leetcode/array/minimum-size-subarray-sum.cpp
#include <iostream>
#include <vector>
using namespace std;

int minSubArrayLen(int t, vector<int>& nums) {
  int l = 0, r = 0, s = 0;
  int vol = nums.size();
  int count = vol + 1;
  
  cout<<s<<" "<<r<<" "<<l<<"\n";
  while (l < vol) {
    if ( r < vol && s < t) {
      s += nums[r];
      r++;
    } else {
      s -= nums[l];
      l++;
    }
    cout<<s<<" "<<r<<" "<<l<<"\n";
    
    if (s >= t && r - l < count) {
      count = r - l;
    }
  }
  return count > vol ? 0 : count;
}

int main() {
  vector<int> n = { 2,3,1,2,4,3 };
  cout<<minSubArrayLen(7, n)<<"\n";
}
